### PR TITLE
[Onboarding] sync schema and add collector profile mutation for collector intent

### PR DESCRIPTION
**Pull request by xtina-starr**
This adds the actual data submission for the [collector intent question for onboarding](https://github.com/artsy/collector-experience/issues/504).

### CODE REVIEW

**Comment by alloy** [comment]
Looks good 👍 What’s still in progress?

**Comment by xtina-starr** [comment]
@alloy nothing really. I just anticipated that there would be feedback 😄

**Comment by alloy** [comment]
Not sure if this PR needs ammending after making [these changes](https://github.com/artsy/metaphysics/pull/817#discussion_r151980566), let’s wait for that first?

**Comment by xtina-starr** [comment]
@alloy I'm trying to map the options that were to select to the enum value that should be sent to MP. I was wondering how do I bring in the enum for MP? So that when "buy art & design" and "learn about art" are selected I can pass `[BUY_ART_AND_DESIGN, LEARN_ABOUT_ART]` as the input value.

**Comment by alloy** [comment]
Hmm, good question. Ideally in the future we can verify the value is one of the enum entries by using TS, but for now I think you’ll just have to pass the values.

**Comment by alloy** [comment]
...meaning that this PR may already be complete. Have you tested if it still works?

**Comment by xtina-starr** [comment]
@alloy it doesn't because metaphysics is expecting it to be one of the enum values (i.e. `BUY_ART_AND_DESIGN`). This is the error:
Uncaught Error: No data returned for operation `CollectorIntentUpdateCollectorProfileMutation`, got error(s):

```
Variable "$input" got invalid value {"intents":{"intents":["buy art & design"]}}.
In field "intents": Expected type "Intents", found {"intents":["buy art & design"]}.
```

**Comment by xtina-starr** [comment]
And then when I hardcode the intents values as `[BUY_ART_AND_DESIGN]` I get and uncaught reference error  because `BUY_ART_AND_DESIGN` isn't defined. Which makes me think I need to some how bring in the enum from metaphysics... or maybe I implemented the enum incorrectly 🤔

File: src/Components/Onboarding/Steps/CollectorIntent.tsx [hunk 1]
```diff
@@ -24,12 +25,12 @@ interface State {
 
 class CollectorIntent extends React.Component<Props, State> {
   options = [
-    "Buy Art & Design",
-    "Sell Art & Design",
-    "Research Art Prices",
-    "Learn About Art",
-    "Find Out About New Exhibitions",
-    "Read Art Market News",
+    "buy art & design",
```
Comment (mzikherman) [inline src/Components/Onboarding/Steps/CollectorIntent.tsx]: Should these be updated to match the enums in Metaphysics?

ie- `"BUY_ART_AND_DESIGN"`

**Comment by orta** [comment]
`BUY_ART_AND_DESIGN` is a variable name when inside JS, it can only be used like that inside a GraphQL query ( for an example see [this test](https://github.com/artsy/emission/blob/master/src/lib/Components/Consignments/Submission/__tests__/objectToGraphQL-tests.ts#L22-L26) )

**Comment by mzikherman** [comment]
Lol we triple posted! I think you just need quotes around that enum value in this context. `"BUY_ART_AND_DESIGN"`

**Comment by xtina-starr** [comment]
So that indeed worked. Thanks @orta and @matt!

**Comment by alloy** [comment]
Dope! 👍

**Pull request merged**